deps/wazuh_testing/wazuh_testing/scripts/statistical_data_analyzer.py

```diff
@@ -0,0 +1,339 @@
+    returns whether there is a significant change based on a threshold value.
+
+    Args:
+        baseline: Dataframe with the baseline data.
+        datasource: Dataframe with the data source.
+        daemon: concrete daemon on which to obtain the values.
+        value: metric from which the statistics to be compared are obtained.
+        stat: concrete statistic to be compared.
+        threshold: Threshold for comparison. If not specified, default is 5%.
+
+    Returns:
+        discrepancie: If the percentage difference is greater than the threshold,
+        it returns 1, otherwise it returns 0.
+    """
+    baseline_statistics = calculate_basic_statistics(baseline)
+    dataframe_statistics = calculate_basic_statistics(datasource)
+    discrepancie = 0
+    baseline_value = get_stat_value(baseline_statistics[daemon], value, stat)
+    dataframe_value = get_stat_value(dataframe_statistics[daemon], value, stat)
+
+    if baseline_value != 0:
+        diff = abs(baseline_value - dataframe_value) / baseline_value
+    else:
+        diff = abs(baseline_value - dataframe_value)
+    
+    if diff >= threshold:
+        discrepancie = 1
+    
+    return discrepancie
+
+
+def t_student_test(baseline, datasource, value):
+    """Function that performs the statistical analysis using the t-student test.
+
+    Args:
+        baseline: Dataframe with the baseline data.
+        datasource: Dataframe with the data source.
+        value: metric on which the test is performed
+
+    Returns:
+        t_p_value: p value returned by the t-student test.
+    """
+    ref_values = baseline[value].dropna()
+    new_values = datasource[value].dropna() 
+
+    t_stat, t_p_value = ttest_ind(ref_values, new_values, equal_var=False)
+
+    return t_p_value  
+
+
+def t_levene_test(baseline, datasource, value):
+    """Function that performs the statistical analysis using the Levene test.
+
+    Args:
+        baseline: Dataframe with the baseline data.
+        datasource: Dataframe with the data source.
+        value: metric on which the test is performed.
+    
+    Returns:
+        l_p_value: p value returned by the Levene test.
+    """
+    ref_values = baseline[value].dropna()
+    new_values = datasource[value].dropna() 
+
+    l_stat, l_p_value = levene(ref_values, new_values)
+
+    return l_p_value
+
+
+def t_anova_test(baseline, datasource, value):
+    """Function that performs the statistical analysis using the ANOVA test.
+
+    Args:
+        baseline: Dataframe with the baseline data.
+        datasource: Dataframe with the data source.
+        value: metric on which the test is performed.
+
+    Returns:
+        a_p_value: p value returned by the ANOVA test.
+    """
+    ref_values = baseline[value].dropna()
+    new_values = datasource[value].dropna() 
+
+    a_stat, a_p_value = f_oneway(ref_values, new_values)
+
+    return a_p_value
+
+
+def data_comparison_test(baseline, datasource, percentage=95):
```

Comment:
This method does not allow getting all the values of the statistics, it only prints a report. This can not be ingested by other functions or tools. It's better to avoid this logic in the library

Reply:
Done in https://github.com/wazuh/wazuh-qa/pull/5622/commits/26dc5cfa8d7f2339d13d5d5820d1ad7cceda3625